SwiftUIBestPractices.md

```diff
@@ -0,0 +1,143 @@
+# SwiftUI Best Practices
```

Comment:
Since we already have a SwiftUI section in the table of contents/Readme, a link to this document should be added there as well. We may want to name this something like "Miscellaneous SwiftUI Best Practices" since we’re already splitting things up over a couple documents, and over time, if these topics within grow, we can move them out to their own documents within the SwiftUI section.

Reply:
We ultimately decided against this, and changed the TOC label beneath the SwiftUI section to be more general, but kept the actual document name and title the same in case it’s read in isolation.